aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/core/handlers/container/common/ContainerHandler.java

```diff
@@ -123,169 +101,136 @@ public void accept(Class<?> componentClass, Element parentElement) {
+            }
+            if (cls.isAnnotationPresent(Dialog.class)) {
+                if (containerName.equals("tabs")) {
+                    Arrays.stream(cls.getAnnotation(Dialog.class).tabs())
+                            .forEach(tab -> {
+                                TabContainerInstance tabContainerInstance = new TabContainerInstance(tab.title());
+                                tabContainerInstance.setAttributes(getAnnotationFields(tab));
+                                result.put(tab.title(), tabContainerInstance);
+                            });
+                } else {
+                    Arrays.stream(cls.getAnnotation(Dialog.class).accordionTabs())
+                            .forEach(tab -> {
+                                TabContainerInstance tabContainerInstance = new TabContainerInstance(tab.title());
+                                tabContainerInstance.setAttributes(getAnnotationFields(tab));
+                                result.put(tab.title(), tabContainerInstance);
+                            });
+                }
+            }
         }
-        return tab.title().equalsIgnoreCase(field.getAnnotation(PlaceOnTab.class).value());
+        return result;
     }
 
-
-    /**
-     * Represents an aggregate of {@link Tab} instance and a list of fields designed to be rendered within this tab.
-     * Used to compose a sorted "tab registry" for a component class
-     * @see TabsHandler#getTabInstances(List)
-     */
-    private static class TabInstance {
-        private Tab tab;
-
-        private List<Field> fields;
-
-        /**
-         * Creates a new {@code TabInstance} wrapped around a specified {@link Tab} with an empty list of associated fields
-         * @param tab {@code Tab} object
-         */
-        private TabInstance(Tab tab) {
-            this.tab = tab;
-            this.fields = new LinkedList<>();
-        }
-
-        /**
-         * Creates a new {@code TabInstance} wrapped around a specified {@link Tab} with a particular list of associated
-         * fields
-         * @param tab {@code Tab} object
-         * @param fields List of {@code Field} objects to associate with the current tab
-         */
-        private TabInstance(Tab tab, List<Field> fields) {
-            this.tab = tab;
-            this.fields = new LinkedList<>(fields);
-        }
-
-        /**
-         * Gets the stored {@link Tab}
-         * @return {@code Tab} object
-         */
-        private Tab getTab() {
-            return tab;
+    default Map<String, Object> getAnnotationFields(Annotation annotation) {
+        try {
+            Object handler = Proxy.getInvocationHandler(annotation);
+            Field f = handler.getClass().getDeclaredField("memberValues");
+            f.setAccessible(true);
+            return (Map<String, Object>) f.get(handler);
+        } catch (Exception ignored) {
+            //ignored
         }
+        return new HashMap<String, Object>();
```

Comment:
Type params are not needed within the diamond operator

Reply:
Fixed